lib/cisco_node_utils/yum.rb

```diff
@@ -32,9 +33,16 @@ def self.decompose_name(file_name)
       # ex: b+z-ip2.x64_64
       name_arch_regex = /^([\w\-\+]+)\.(\w+)$/
 
-      file_name.match(name_ver_arch_regex) ||
-        file_name.match(name_ver_arch_regex_nx) ||
-        file_name.match(name_arch_regex)
+      # xrv9k-k9sec-1.0.0.0-r61102I.x86_64.rpm-XR-DEV-16.02.22C
+      name_arch_regex_xr = /^(.*\d.*)-([\d.]*)-(r\d\d\d\d*.).(\w{4,}).rpm/
```

Comment:
Looking at the 6.0.0 release, it looks like release RPMs may follow the format `'xrv9k-k9sec-1.0.0.0-r600.x86_64.rpm-6.0.0'` which your match group `(r\d\d\d\d*.)` would fail to match against. Also, you should escape the literal `.`s before `rpm` and before the architecture match group. Perhaps the below is preferable:

`
+      name_arch_regex_xr = /^(.*\d.*)-([\d.]*)-(r\d+.*)\.(\w{4,})\.rpm/`